File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int first = -1, last = -1, flag =0;
        vector<int>ans;
        for(int i =0 ; i < nums.size(); i++){
           if(nums[i] == target && flag ==0){
            first = i ; 
            last = i; flag =1;}
            if(nums[i] == target && i > last && flag == 1){
                last = i;
            }
           
        }
        ans.push_back(first); ans.push_back(last);
        return ans;
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        
        int st = 0, end = nums.length - 1;
        int ans1 =-1, ans2 = -1;
        while(st<=end){
            int mid  = (st + end)/2;

            if(nums[mid] == target){
            ans1 = mid;
            end = mid-1;
            }
            else if(nums[mid] > target) end = mid-1;
            else st = mid+1;
        }
        st = 0; end = nums.length - 1;
         while(st<=end){
            int mid  = (st + end)/2;

            if(nums[mid] == target){
            ans2 = mid;
            st = mid+1;
            }
            else if(nums[mid] < target)st = mid+1;
            else end = mid-1;
        }

        return new int[]{ans1,ans2};

        
    }
}